.github/workflows/build.yml

```diff
@@ -4,98 +4,32 @@ on:
   push:
     branches:
       - master
-  pull_request:
-    types:
-      - opened
-      - synchronize
-      - reopened
+  workflow_call:
+    inputs:
+      configuration:
+        type: string
+        default: Debug
 
 jobs:
   build:
     runs-on: ubuntu-latest
-
-    strategy:
-      fail-fast: false
-      matrix:
-        mono:
-          - latest
-          - '6.12'
-          - '6.10'
-          - '6.8'
-        configuration:
-          - Debug
-          - Release
-
-    container:
-      image: mono:${{ matrix.mono }}
-
     steps:
-      - name: Adding HTTPS support to APT for old Mono images
-        run: |
-          apt-get update || true
-          apt-get install -y apt-transport-https
-      - name: Installing checkout/build dependencies
-        run: apt-get update && apt-get install -y git
-      - uses: actions/checkout@v3
-
+      - uses: actions/checkout@v4
       - name: Setup .NET Core
-        uses: actions/setup-dotnet@v3
+        uses: actions/setup-dotnet@v4
         with:
           dotnet-version: '7'
```

Comment:
Any thoughts on removing this "Setup .NET Core" step, since we discovered that the dotnet tools are already installed in the container by default?
(and ".NET Core" isn't even the name anymore)

Reply:
I tried that first and it complained. And whilst they are installed in the actions container by default, they *not* installed in the mono container.

Reply:
What was the complaint? Remember, it's a no-op as is:

![image](https://github.com/KSP-CKAN/CKAN/assets/1559108/061b9cc6-ee9c-4613-8d40-0e5fe268f3a2)

Reply:
Lemme push to re-check the assumption, it might be a path / job setup thing. If it's basically a no-op there are no consequences for leaving it in, though it does add a point of maintenance, so if it's just pathing maybe we can export an path or something.

Reply:
Pathing probably
```
Run xvfb-run ./build test+only --configuration=Debug --where="Category!=FlakyNetwork"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 8224k  100 8224k    0     0   125M      0 --:--:-- --:--:-- --:--:--  125M
./build: 43: ./build: dotnet: not found
./build: 64: ./build: dotnet: not found
```

Reply:
It's possible these tests aren't really super relevant, but it's definitely not a no-op inside the containers

https://github.com/KSP-CKAN/CKAN/actions/runs/8949111709/job/24583068605#step:7:1

What are we testing for inside these containers? Is it even relevant? Can we actually test different mono versions?

Happy to make an adjustment, but when it came to the test cases, I mostly tidied them up and brought them across with the existing configuration.

Reply:
Wait. Why is a change to `build.yml` affecting `test.yml`? I thought the only things that transfer over were the artifacts?

Reply:
... oh! You updated `test.yml` instead! Yeah, that's definitely not what I was suggesting. I'll fix that...

Reply:
Looks like this came back in the latest changes. Note to self to purge it again.

Reply:
Done.